core/tee/socket.c

```diff
@@ -225,11 +225,10 @@ static TEE_Result pta_socket_open_session(uint32_t param_types __unused,
 			TEE_Param pParams[TEE_NUM_PARAMS] __unused,
 			void **sess_ctx)
 {
-	struct tee_ta_session *s;
+	struct ts_session *s = ts_get_calling_session();
 
 	/* Check that we're called from a TA */
-	s = tee_ta_get_calling_session();
-	if (!s)
+	if (!is_user_ta_ctx(s->ctx))
```

Comment:
If this PTA is called from REE, `ts_get_calling_session()` will return `NULL` hence a segfault here.
Maybe replace with `if (!s || !is_user_ta_ctx(s->ctx))`.

Reply:
comment still applies (I think).

Reply:
I'm sorry, I missed this. I'll fix.